runtime/src/main/java/org/corfudb/runtime/view/workflows/WorkflowRequest.java

```diff
@@ -130,19 +142,20 @@ private void waitForWorkflow(@NonNull UUID workflow, @NonNull ManagementClient c
      * verified.
      */
     public void invoke() {
-        for (int x = 0; x < retry; x++) {
+        for (int retryIteration = 0; retryIteration < retry; retryIteration++) {
             try {
                 Optional<ManagementClient> orchestrator = getOrchestrator();
                 if(orchestrator.isPresent()){
                     UUID workflowId = sendRequest(orchestrator.get());
                     waitForWorkflow(workflowId, orchestrator.get(), timeout, pollPeriod);
                 }
-                else{
-                    throw new IllegalStateException("Orchestrator can not be selected");
+                else {
+                    log.warn("Orchestrator can not be selected");
```

Comment:
You may need to add the number of attemp in the log, same as what it does when we catch the exception.

Reply:
Noted. Thanks